### PR TITLE
feat: [M3-6779] - AGLB api-v4 endpoints

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝
- Adds api-v4 endpoints and types for the AGLB

## Major Changes 🔄
- Adds all endpoints that should be needed for the AGLB project

## How to test 🧪
- No testing to be done, just review the endpoint, spelling, and typescript types
  - When reviewing, compare type to the API specification linked on the internal ticket

### CODE REVIEW

File: packages/api-v4/src/aglb/types.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  tags: string[];
+  name: string;
+  regions: string[];
+  'entry-points': string[];
+}
+
+export interface CreateLoadbalancerPayload {
+  name: string;
+  regions: string[];
+  entrypoints: EntrypointPayload[];
```
Comment (jdamore-linode) [inline packages/api-v4/src/aglb/types.ts:12]: I know this isn't on us, but wondering if we could get the API team to improve the consistency between `entry-points` on `Loadbalancer` and `entrypoints` on `CreateLoadbalancerPayload`.

The casing on these properties seems inconsistent on a broader level: some properties are `camelCase` (`CertificateTable.sniHostname`, for example), some are `snake_case` (which I think is generally the default?), and then there's this one `kebab-case` property. Do we think the API team might be willing to settle on one style and clean this up, or is there more to this?

**Review by jdamore-linode — APPROVED** [review]

File: packages/api-v4/src/aglb/types.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  name: string;
+  regions: string[];
+  entrypoints: string[];
+}
+
+export interface CreateLoadbalancerPayload {
+  name: string;
+  regions: string[];
+  entrypoints: EntrypointPayload[];
+}
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/types.ts:13]: Just confirming: we're excluding `tags` in this CreateLoadbalancePayload and the UpdateLoadbalancerPayload?

**Review by mjac0bs — APPROVED** [review]

File: packages/api-v4/src/aglb/types.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+export interface Match {
+  path: string;
+}
+
+export interface ServiceTargetPayload {
+  name: string;
+  endpoints: Endpoint[];
+  ca_certificate: string;
+  load_bal_policy: Policy;
```
Comment (mjac0bs) [inline packages/api-v4/src/aglb/types.ts:70]: API has this as `load_balancing_policy`, not sure if something has changed.

**Comment by jaalah-akamai** [comment]
@bnussman-akamai this looks good pending your questions in the API spec, maybe we should add comments next to the things we're waiting to hear back from as to not forget about them 👍

**Review by jaalah-akamai — APPROVED** [review]

File: packages/api-v4/src/aglb/types.ts (hunk 2 above)
Comment (bnussman-akamai) [reply packages/api-v4/src/aglb/types.ts:13]: Added `tags`!

File: packages/api-v4/src/aglb/types.ts (hunk 3 above)
Comment (bnussman-akamai) [reply packages/api-v4/src/aglb/types.ts:70]: API spec had both... `load_balancing_policy ` is correct. Fixed!

**Pull request merged**